### PR TITLE
✅🐳 Update Dockerfile. Refactor test start strategy

### PR DESCRIPTION
**Pull request by dominikkaminski**
**Changes:**

1. Update Dockerfile to latest node version
2. Add missing dependencies in Dockerfile
3. Make it easier to start protractor tests
4. Start Docker container automatically when using `npm test-init && npm test-docker`
5. Change `fdescribe` to `describe`

PR: #898 

## How can others test the changes?

```shell
npm run test-init
npm run test
```

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.).
- [x] I've rebased the `develop` branch with my branch before finishing this PR.
- [x] I've **summarized all changes** in a list above.
- [x] I've mentioned all **PRs, which relate to this one**.
- [x] I've prefixed my Pull Request title is according to [gitmoji guide](https://gitmoji.carloscuesta.me/).

## Example

<details>
<summary>
:warning: Before merging please click here to expand and follow the guide.
</summary>
<br>

Please use `:twisted_rightwards_arrows:` at the beginning of your merge commit title.

Example 1:

<pre>
<code>
:twisted_rightwards_arrows: :bug: Fix Wrong Text Decoration at ...
</code>
</pre>

To get your commit message, just copy the first part of this pull request.

Example 2:
<pre>
<code>
**Changes:**

- Fixes Wrong Text Decoration at ...
- Fixes some typos
- ...

**Issues:**

Closes #NumberOfFixedIssue

PR: #NumberOfThisPR
</code>
</pre>
</details>

## Emojis

<details>
<summary>
Expand for a list of most used Emojis.
</summary>
<br>

Please prefix your commit messages with an Emoji.

Ref: https://gitmoji.carloscuesta.me/

| Description              | Glyphe               | Emoji  |
|--------------------------|----------------------|--------|
| Bugfix                   | `:bug:`              | 🐛     |
| Configuration releated   | `:wrench:`           | 🔧     |
| Cosmetic                 | `:lipstick:`         | 💄     |
| Dependencies Downgrade   | `:arrow_down:`       | ⬇️     |
| Dependencies Upgrade     | `:arrow_up:`         | ⬆️     |
| Formatting               | `:art:`              | 🎨     |
| Improving Performance    | `:zap:`              | ⚡️     |
| Initial commit           | `:tada:`             | 🎉     |
| Linter                   | `:rotating_light:`   | 🚨     |
| Miscellaneous            | `:package:`          | 📦     |
| New Feature              | `:sparkles:`         | ✨     |
| Refactoring Code         | `:recycle:`          | ♻️     |
| Releasing / Version tags | `:bookmark:`         | 🔖     |
| Removing Stuff           | `:fire:`             | 🔥     |
| Tests                    | `:white_check_mark:` | ✅     |
| Work In Progress (WIP)   | `:construction:`     | 🚧     |

</details>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -236,56 +232,35 @@ Es werden folgende Pakete benötigt:
...
-   npm run integration-test-init
+   npm run test-init
    ```
 
 1. Die End-to-End tests können in einem weiteren Terminal mit folgendem Befehl
    gestartet werden:
 
    ```shell
-   npm run integration-test
+   npm run test
```
Comment (robpal94) [inline README.md]: We could actually user `npm test` here.

File: test/start_docker_bundle.sh [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+
+name='bpmn-studio'
+
+[[ $(docker ps -f "name=$name" --format '{{.Names}}') == $name ]] ||
+docker run --name $name -d -p 8000:8000 -p 9000:9000 5minds/bpmn-studio-bundle:develop
```
Comment (robpal94) [inline test/start_docker_bundle.sh]: This may be a bit more _reader-friendly_: 
```bash
if [[ $(docker ps -f "name=$name" --format '{{.Names}}') != "$name"  ]]; then
    docker run --name "$name" -d -p 8000:8000 -p 9000:9000 5minds/bpmn-studio-bundle:develop 
fi
```

File: test/start_docker_bundle.sh [hunk 3]
```diff
@@ -2,5 +2,6 @@
 
 name='bpmn-studio'
 
-[[ $(docker ps -f "name=$name" --format '{{.Names}}') == $name ]] ||
-docker run --name $name -d -p 8000:8000 -p 9000:9000 5minds/bpmn-studio-bundle:develop 
+if [[ $(docker ps -f "name=$name" --format '{{.Names}}') != "$name"  ]]; then
```
Comment (robpal94) [inline test/start_docker_bundle.sh]: trailing whitespace after `!= "$name"`

File: README.md [hunk 4]
```diff
@@ -236,56 +232,35 @@ Es werden folgende Pakete benötigt:
...
-1. Ein weiteres Terminal öffnen und folgenden Befehl eingeben, um den Selenium
-   Server zu starten:
+1. Anschließend folgenden Befehl eingeben, um den Selenium Server und die ProcessEngine zu starten:
 
    ```shell
-   npm run integration-test-init
+   npm run test-init
    ```
 
 1. Die End-to-End tests können in einem weiteren Terminal mit folgendem Befehl
```
Comment (alexanderkasten) [inline README.md]: Tests

File: package.json [hunk 5]
```diff
@@ -120,6 +121,7 @@
     "@fortawesome/fontawesome-free": "^5.1.0",
     "@process-engine/bpmn-js-custom-bundle": "^2.0.0",
     "@process-engine/process_engine_contracts": "^7.0.0",
+    "@process-engine/process_engine_runtime": "latest",
```
Comment (robpal94) [inline package.json]: Why is this here ?
/cc @Paulomart

Comment (dominikkaminski) [reply package.json]: @robpal94 It is there because you need a ProcessEngine to use End-to-End tests.

Comment (robpal94) [reply package.json]: But we already import it [in line 91](https://github.com/process-engine/bpmn-studio/pull/898/files#diff-b9cfc7f2cdf78a7f4b91a753d10865a2L91)

File: aurelia_project/tasks/e2eDocker.ts [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+import * as tsConfig from './../../tsconfig.json';
+import * as project from './../aurelia.json';
+
+function clean(): Promise<Array<string>> {
+  return del(`${project.e2eTestRunner.dist}/*`);
+}
+
+function build_tests(): NodeJS.ReadWriteStream {
+
+  const typescriptCompiler: typescript.Project = typescript.createProject(Object.assign({}, tsConfig.compilerOptions, {
```
Comment (robpal94) [inline aurelia_project/tasks/e2eDocker.ts]: Please remove this method chain like so: 

```
const compilerOptionsCopy: SomeType = Object.assign({}, tsConfig.compilerOptions, {
    module: 'commonjs',
  });
const typescriptCompiler: typescript.Project = typescript.createProject(compilerOptionsCopy)
```

File: test/start_docker_bundle.sh [hunk 7]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+name='bpmn-studio'
+
+if [[ $(docker ps -f "name=$name" --format '{{.Names}}') != "$name" ]]; then
+    docker run --name "$name" -d -p 8000:8000 -p 9000:9000 5minds/bpmn-studio-bundle:latest 
+else
+    docker rm -v -f "$name"
+    docker run --name "$name" -d -p 8000:8000 -p 9000:9000 5minds/bpmn-studio-bundle:latest
```
Comment (robpal94) [inline test/start_docker_bundle.sh]: Maybe you could format this line a bit nicer: 

```
docker run\
    --name "$name"\
    -d\
    -p 8000:8000\
    -p 9000:9000\
    5minds/bpmn-studio-bundle:latest
```
